### PR TITLE
[EM-820][News] add News to referent space

### PR DESCRIPTION
**Pull request by elvino29**

### CODE REVIEW

File: templates/jecoute/news/_form.html.twig [hunk 1]
```diff
@@ -22,13 +22,15 @@
     {{ form_errors(form.zone) }}
 </div>
 
+{% if isNotifiable %}
```
Comment (Remg) [inline templates/jecoute/news/_form.html.twig:25]: 1/ twig var names should be in snake case => `is_notifiable` (not mandatory)

2/ can't you retrieve it from the form options ? 
e.g. : add the option in the `buildView` method of the form type, then retrieve it with `form.vars.is_notifiable`

File: src/Security/Voter/ReferentNewsPublicationVoter.php [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+namespace App\Security\Voter;
+
+use App\Entity\Adherent;
+use App\Entity\Jecoute\News;
+use App\Repository\Geo\ZoneRepository;
+
+class ReferentNewsPublicationVoter extends AbstractAdherentVoter
+{
+    public const PERMISSION = 'IS_ABLE_TO_PUBLISH';
```
Comment (Remg) [inline src/Security/Voter/ReferentNewsPublicationVoter.php:11]: can you make the role name more specific to Jecoute\News ? 
able ==> allowed

File: src/Controller/EnMarche/Jecoute/News/AbstractNewsController.php [hunk 3]
```diff
@@ -61,8 +62,13 @@ public function jecouteNewsCreateAction(
             $news->setZone($zones[0]);
         }
 
+        $options = ['zones' => $zones];
+        if (JecouteSpaceEnum::REFERENT_SPACE === $this->getSpaceName()) {
+            $options['is_notifiable'] = false;
+        }
```
Comment (ottaviano) [inline src/Controller/EnMarche/Jecoute/News/AbstractNewsController.php:68]: what about adding a new abstract method: `isNotifiableNews`?

```suggestion
        $options = [
          'zones' => $zones,
          'is_notifiable' => $this->isNotifiableNews(),
        ];
```

File: src/Security/Voter/ReferentNewsPublicationVoter.php [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    public function __construct(ZoneRepository $zoneRepository)
+    {
+        $this->zoneRepository = $zoneRepository;
+    }
+
+    protected function doVoteOnAttribute(string $attribute, Adherent $adherent, $subject): bool
+    {
+        $zones = $this->zoneRepository->findForJecouteByReferentTags($adherent->getManagedArea()->getTags()->toArray());
+
+        return $adherent->isReferent() && \in_array($subject->getZone(), $zones, true);
```
Comment (ottaviano) [inline src/Security/Voter/ReferentNewsPublicationVoter.php:24]: could you check `$adherent->isReferent()` before calling `$adherent->getManagedArea()->getTags()`. 
`$adherent->getManagedArea()` returns `null` if user does not have referent role.

**Review by ottaviano — APPROVED** [review]

**Review by Remg — APPROVED** [review]

**Pull request merged**